File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Float}
def find_max_average(nums, k)
    sum=nums.slice(0,k).sum
    avg=sum.to_f/k
    for i in k..nums.length-1
        sum-=nums[i-k]
        sum+=nums[i]
         if sum/k.to_f>avg
             avg=sum.to_f/k
         end
    end
    avg
end

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let sum=0
    for(let i=0;i<k;i++){
        sum+=nums[i]
    }
    let maxSum=sum
    for(let i=k;i<nums.length;i++){
       console.log(i,k,sum,nums[i])
        sum=(sum-nums[i-k]) + nums[i]
        if(sum>maxSum)
            maxSum=sum
    }
    
    return maxSum/k
};